File: src/2025-03-04-12-20-21--deepseek-vr-0304/java/exercises/practice/food-chain/src/main/java/FoodChain.java
class FoodChain {

    private static final String[] ANIMALS = {
        "fly",
        "spider",
        "bird",
        "cat",
        "dog",
        "goat",
        "cow",
        "horse"
    };

    private static final String[] COMMENTS = {
        "",
        "It wriggled and jiggled and tickled inside her.",
        "How absurd to swallow a bird!",
        "Imagine that, to swallow a cat!",
        "What a hog, to swallow a dog!",
        "Just opened her throat and swallowed a goat!",
        "I don't know how she swallowed a cow!",
        ""
    };

    String verse(int verse) {
        if (verse == 8) {
            return "I know an old lady who swallowed a horse.\nShe's dead, of course!";
        }

        StringBuilder sb = new StringBuilder();
        sb.append("I know an old lady who swallowed a ")
            .append(ANIMALS[verse - 1])
            .append(".\n");

        if (verse > 1) {
            sb.append(COMMENTS[verse - 1]).append("\n");
        }

        for (int j = verse - 1; j >= 1; j--) {
            sb.append("She swallowed the ")
                .append(ANIMALS[j])
                .append(" to catch the ")
                .append(ANIMALS[j - 1]);
            if (j - 1 == 1) { // When previous animal is spider
                sb.append(" that wriggled and jiggled and tickled inside her.");
            } else {
                sb.append(".");
            }
            sb.append("\n");
        }

        sb.append("I don't know why she swallowed the fly. Perhaps she'll die.\n");

        return sb.toString();
    }

    String verses(int startVerse, int endVerse) {
        StringBuilder sb = new StringBuilder();
        for (int i = startVerse; i <= endVerse; i++) {
            sb.append(verse(i));
            if (i != endVerse) {
                sb.append("\n");
            }
        }
        return sb.toString();
    }
}

File: src/2025-03-04-12-20-21--deepseek-vr-0304/go/exercises/practice/food-chain/food_chain.go
package foodchain

import (
	"strings"
)

type animal struct {
	name     string
	intro    string
	action   string
	previous *animal
}

var animals = []animal{
	{"fly", "I know an old lady who swallowed a fly.", "I don't know why she swallowed the fly. Perhaps she'll die.", nil},
	{"spider", "I know an old lady who swallowed a spider.", "It wriggled and jiggled and tickled inside her.", nil},
	{"bird", "I know an old lady who swallowed a bird.", "How absurd to swallow a bird!", nil},
	{"cat", "I know an old lady who swallowed a cat.", "Imagine that, to swallow a cat!", nil},
	{"dog", "I know an old lady who swallowed a dog.", "What a hog, to swallow a dog!", nil},
	{"goat", "I know an old lady who swallowed a goat.", "Just opened her throat and swallowed a goat!", nil},
	{"cow", "I know an old lady who swallowed a cow.", "I don't know how she swallowed a cow!", nil},
	{"horse", "I know an old lady who swallowed a horse.", "She's dead, of course!", nil},
}

func init() {
	// Set up previous pointers
	for i := 1; i < len(animals); i++ {
		animals[i].previous = &animals[i-1]
	}
}

func Verse(v int) string {
	if v < 1 || v > 8 {
		return ""
	}
	
	a := animals[v-1]
	var lines []string
	
	// Add the intro lines
	lines = append(lines, a.intro)
	
	// Special case for horse
	if a.name == "horse" {
		lines = append(lines, a.action)
		return strings.Join(lines, "\n")
	}
	
	// Add the action line for current animal
	lines = append(lines, a.action)
	
	// Build the swallowed chain
	current := &a
	for current.previous != nil {
		prev := current.previous
		line := "She swallowed the " + current.name + " to catch the " + prev.name
		if prev.name == "spider" {
			line += " that wriggled and jiggled and tickled inside her."
		} else {
			line += "."
		}
		lines = append(lines, line)
		current = current.previous
	}
	
	// Add the ending lines if not fly
	if a.name != "fly" {
		lines = append(lines, animals[0].action)
	}
	
	return strings.Join(lines, "\n")
}

func Verses(start, end int) string {
	var verses []string
	for i := start; i <= end; i++ {
		verses = append(verses, Verse(i))
	}
	return strings.Join(verses, "\n\n")
}

func Song() string {
	return Verses(1, 8)
}
